torch/fx/experimental/proxy_tensor.py

```diff
@@ -104,25 +104,34 @@ def set_meta(proxy, val):
             proxy.node.meta['tensor_meta'] = _extract_tensor_metadata(val)
     return proxy
 
+def thunkify(f, *args, **kwargs):
+    """
+    Delays computation of f until it's called again, which is done by
+    """
+    return functools.lru_cache(1)(functools.partial(f, *args, **kwargs))
```

Comment:
lru_cache boooooo (can be faster if done manually)

Reply:
If you remember, I used to manually implement it. But I couldn't benchmark a difference in my testing lol.